### PR TITLE
Add new test job for serving operator

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
As requested by @houshengbo , add a new custom job for serving-operator to run test against the latest serving. 

Bonus:
Use default setting, instead of `fatal` in testgrd config generator, which makes us easier to add a custom job with any name.

/cc @adrcunha 
/cc @chaodaiG 
/cc @houshengbo

### CODE REVIEW

File: ci/prow/config_knative.yaml [hunk 1]
```diff
@@ -172,6 +172,10 @@ presubmits:
       dot-dev: true
     - go-coverage: true
       dot-dev: true
+    - custom-test: integration-tests-on-latest-serving
```
Comment (houshengbo) [inline ci/prow/config_knative.yaml]: We do not need this custom job for PR build so far, only the one in the periodic job is needed. Thx.

File: ci/prow/config_knative.yaml [hunk 2]
```diff
@@ -172,6 +172,10 @@ presubmits:
...
+      always_run: false
+      optional: true
+      full-command: "./test/e2e-tests-latest-serving.sh"
```
Comment (adrcunha) [inline ci/prow/config_knative.yaml]: Don't use full-command, you're skipping all the presubmit checks.

Comment (chizhg) [reply ci/prow/config_knative.yaml]: Done, changed to command

Comment (adrcunha) [reply ci/prow/config_knative.yaml]: It should be args, including a `--run-tests`, just like serving does.

Comment (chizhg) [reply ci/prow/config_knative.yaml]: Changed presubmit job to that format. Also tried to change the periodic job, but seems it's not supported.

File: ci/prow/config_knative.yaml [hunk 3]
```diff
@@ -357,6 +363,14 @@ periodics:
     - auto-release: true
       dot-dev: true
       go113: true
+    - custom-job: integration-tests-on-latest-serving
+      command:
+      - "./test/presubmit-tests.sh"
```
Comment (adrcunha) [inline ci/prow/config_knative.yaml]: This is not a presubmit job, don't use presubmit-tests.sh.

Comment (chizhg) [reply ci/prow/config_knative.yaml]: Sorry got the two messed, fixed.

**Pull request merged**